// @fern-api/openai/serialization/resources/fineTune/client/requests/CreateFineTuneRequest.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateFineTuneRequest = void 0;
const core = __importStar(require("../../../../../core"));
exports.CreateFineTuneRequest = core.serialization.object({
    trainingFile: core.serialization.property("training_file", core.serialization.string()),
    validationFile: core.serialization.property("validation_file", core.serialization.string().optional()),
    model: core.serialization.string().optional(),
    nEpochs: core.serialization.property("n_epochs", core.serialization.number().optional()),
    batchSize: core.serialization.property("batch_size", core.serialization.number().optional()),
    learningRateMultiplier: core.serialization.property("learning_rate_multiplier", core.serialization.number().optional()),
    promptLossWeight: core.serialization.property("prompt_loss_weight", core.serialization.number().optional()),
    computeClassificationMetrics: core.serialization.property("compute_classification_metrics", core.serialization.boolean().optional()),
    classificationNClasses: core.serialization.property("classification_n_classes", core.serialization.number().optional()),
    classificationPositiveClass: core.serialization.property("classification_positive_class", core.serialization.string().optional()),
    classificationBetas: core.serialization.property("classification_betas", core.serialization.list(core.serialization.number()).optional()),
    suffix: core.serialization.string().optional(),
});
