// @fern-api/openai/Client.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OpenAIClient = void 0;
const Client_1 = require("./api/resources/answer/client/Client");
const Client_2 = require("./api/resources/audio/client/Client");
const Client_3 = require("./api/resources/chat/client/Client");
const Client_4 = require("./api/resources/classification/client/Client");
const Client_5 = require("./api/resources/completion/client/Client");
const Client_6 = require("./api/resources/edit/client/Client");
const Client_7 = require("./api/resources/embedding/client/Client");
const Client_8 = require("./api/resources/engine/client/Client");
const Client_9 = require("./api/resources/file/client/Client");
const Client_10 = require("./api/resources/fineTune/client/Client");
const Client_11 = require("./api/resources/image/client/Client");
const Client_12 = require("./api/resources/model/client/Client");
const Client_13 = require("./api/resources/moderation/client/Client");
class OpenAIClient {
    options;
    constructor(options) {
        this.options = options;
    }
    _answer;
    get answer() {
        return (this._answer ??= new Client_1.Answer(this.options));
    }
    _audio;
    get audio() {
        return (this._audio ??= new Client_2.Audio(this.options));
    }
    _chat;
    get chat() {
        return (this._chat ??= new Client_3.Chat(this.options));
    }
    _classification;
    get classification() {
        return (this._classification ??= new Client_4.Classification(this.options));
    }
    _completion;
    get completion() {
        return (this._completion ??= new Client_5.Completion(this.options));
    }
    _edit;
    get edit() {
        return (this._edit ??= new Client_6.Edit(this.options));
    }
    _embedding;
    get embedding() {
        return (this._embedding ??= new Client_7.Embedding(this.options));
    }
    _engine;
    get engine() {
        return (this._engine ??= new Client_8.Engine(this.options));
    }
    _file;
    get file() {
        return (this._file ??= new Client_9.File_(this.options));
    }
    _fineTune;
    get fineTune() {
        return (this._fineTune ??= new Client_10.FineTune(this.options));
    }
    _image;
    get image() {
        return (this._image ??= new Client_11.Image(this.options));
    }
    _model;
    get model() {
        return (this._model ??= new Client_12.Model(this.options));
    }
    _moderation;
    get moderation() {
        return (this._moderation ??= new Client_13.Moderation(this.options));
    }
}
exports.OpenAIClient = OpenAIClient;
