// @fern-api/openai/dist/api/resources/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.moderation = exports.model = exports.image = exports.fineTune = exports.file = exports.engine = exports.embedding = exports.edit = exports.completion = exports.commons = exports.classification = exports.chat = exports.audio = exports.answer = void 0;
exports.answer = __importStar(require("./answer"));
__exportStar(require("./answer/types"), exports);
exports.audio = __importStar(require("./audio"));
__exportStar(require("./audio/types"), exports);
exports.chat = __importStar(require("./chat"));
__exportStar(require("./chat/types"), exports);
exports.classification = __importStar(require("./classification"));
__exportStar(require("./classification/types"), exports);
exports.commons = __importStar(require("./commons"));
__exportStar(require("./commons/types"), exports);
exports.completion = __importStar(require("./completion"));
__exportStar(require("./completion/types"), exports);
exports.edit = __importStar(require("./edit"));
__exportStar(require("./edit/types"), exports);
exports.embedding = __importStar(require("./embedding"));
__exportStar(require("./embedding/types"), exports);
exports.engine = __importStar(require("./engine"));
__exportStar(require("./engine/types"), exports);
exports.file = __importStar(require("./file"));
__exportStar(require("./file/types"), exports);
exports.fineTune = __importStar(require("./fineTune"));
__exportStar(require("./fineTune/types"), exports);
exports.image = __importStar(require("./image"));
__exportStar(require("./image/types"), exports);
exports.model = __importStar(require("./model"));
__exportStar(require("./model/types"), exports);
exports.moderation = __importStar(require("./moderation"));
__exportStar(require("./moderation/types"), exports);
__exportStar(require("./commons/errors"), exports);
__exportStar(require("./answer/client/requests"), exports);
__exportStar(require("./audio/client/requests"), exports);
__exportStar(require("./chat/client/requests"), exports);
__exportStar(require("./completion/client/requests"), exports);
__exportStar(require("./edit/client/requests"), exports);
__exportStar(require("./embedding/client/requests"), exports);
__exportStar(require("./file/client/requests"), exports);
__exportStar(require("./fineTune/client/requests"), exports);
__exportStar(require("./image/client/requests"), exports);
__exportStar(require("./moderation/client/requests"), exports);
